pydm_alarm_plugin/alarm_plugin.py

```diff
@@ -0,0 +1,113 @@
+import logging
+import os
+from kafka.consumer.fetcher import ConsumerRecord
+from pydm.data_plugins.plugin import PyDMPlugin, PyDMConnection
+from pydm.utilities import remove_protocol
+from pydm.widgets.base import PyDMWidget
+from pydm.widgets.channel import PyDMChannel
+from qtpy.QtCore import QObject, QThread
+from slam.alarm_item import AlarmSeverity
+from slam.kafka_reader import KafkaReader
+from typing import Optional
+
+
+logger = logging.getLogger(__name__)
+kafka_logger = logging.getLogger('kafka')
+kafka_logger.setLevel('WARNING')  # Anything less results in too much noise
+
+
+class Connection(PyDMConnection):
+    """
+    A PyDMConnection for sending alarm severity signals to all listeners which have been registered with it.
+    """
+    def __init__(self, channel: PyDMChannel, address: str, protocol: Optional[str] = None,
+                 parent: Optional[QObject] = None):
+        super().__init__(channel, address, protocol, parent)
+        self.add_listener(channel)
+
+    def send_alarm_data(self, severity: AlarmSeverity):
+        """
+        Sends the updated alarm severity to all listeners of this connection.
+
+        Parameters
+        ----------
+        severity : AlarmSeverity
+            The updated severity which will be emitted as the new_severity_signal
+        """
+        if severity is AlarmSeverity.OK:
+            self.new_severity_signal.emit(PyDMWidget.ALARM_NONE)
+        elif severity is AlarmSeverity.MINOR:
+            self.new_severity_signal.emit(PyDMWidget.ALARM_MINOR)
+        elif severity is AlarmSeverity.MAJOR:
+            self.new_severity_signal.emit(PyDMWidget.ALARM_MAJOR)
+        elif severity is AlarmSeverity.INVALID:
+            self.new_severity_signal.emit(PyDMWidget.ALARM_INVALID)
+        elif severity is AlarmSeverity.UNDEFINED:
+            self.new_severity_signal.emit(PyDMWidget.ALARM_DISCONNECTED)
+
+
+class AlarmPlugin(PyDMPlugin):
+    """
+    Manages the data flow between the kafka queue for alarm data and the PyDM display widgets. Currently this is
+    read-only as no write actions can be taken from PyDM widgets for now.
+    """
+    protocol = "alarm"
```

Comment:
A `nalms://` protocol might be a bit more specific/apt?

Reply:
Good idea, that does make it clearer 492e38b7174cd333a294f4434e65ff4de923e5eb